src/octoprint/plugins/logging/static/js/logging.js

```diff
@@ -104,10 +97,15 @@ $(function () {
                 levels.push(item);
                 configuredLoggers.push(logger);
             });
+            let sortedLevels = _.sortBy(levels, function (o) {
+                return o.component;
+            });
```

Comment:
The sort is moved down to here and only happens when the settings dialog is first opened.

The reason is because it is inconvenient for the logging level to get sorted right after you add a new one.  For example, if you add the sixth logging level and then it sorts, it might jump to row, say, 3.  If you want to now change it from `WARNING` to `DEBUG`, you have to find it again in the list.

That's annoying and the user will not expect it.  So we'll only sort when the logging levels are retrieved from the server, which happens every time the setting dialog is opened.